content/lib/2021-04-02-operating-system-interface.md

```diff
@@ -0,0 +1,225 @@
+---
+title: "The Distributed Operating System Void"
+date: "2021-04-02"
+author: "kris nóva"
+keywords: ""
+summary: ""
+---
+
+_Preface: I will be typing this in proper (capitalized) English. 
+Unlike the majority of my other work. 
+A substantial portion of this is IP is accredited to the concepts proposed by Andrew Rynhard and his work with Talos Systems. (COSi)_ 
+
+## Thesis
+
+Modern (circa 2021) UNIX based operating systems should be reimagined to support higher level workloads 
+that are managed in a distributed environment, such as kubernetes.
+
+The first step in progress towards this goal, is defining an interface.
+
+The Distributed Operating System Interface (`DOSi`) hopes to achieve this.
+
+# Problem Statement 
+
+Over the past 7 years Kubernetes has grown in popularity. 
+Kubernetes is built and managed on operating systems optimized for isolated workloads.
+Kubernetes mutates operating system level features, concerns, and services.
+Some of these services (CNI, Kubelet, Container Runtime, etc) are paramount for Kubernetes to run.
+
+I believe there should be a clear communication and management interface between Kubernetes and the underlying operating system.
+
+This interface does not exist, therefore an operating system implementation does not exist.
+
+# Vocabulary 
+
+#### Distributed Operating System Interface
```

Comment:
I wanted to pick something other than `cosi` mostly because of the conversation we had about going through the exercise of "what would this look like if i wrote it from scratch". Which meant (to me anyway) another name. 

FWIW I also like `dos` interface mostly because this will mean i might be able to one day say the phrase:

 > in 2021 i helped re-invent DOS after a global pandemic.